dm/config/source_config.go

```diff
@@ -318,3 +318,18 @@ func (c *SourceConfig) check(metaData *toml.MetaData, err error) error {
 	c.adjust()
 	return nil
 }
+
+// PreCheck check valify source config
+func (c *SourceConfig) PreCheck(db *sql.DB) error {
```

Comment:
`Adjust` could check and return error, so maybe we could put these code into `Adjust`?

Reply:
agree. seems we have use `*sql.DB` in `Adjust` now.